consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -0,0 +1,171 @@
+ * {@code HealthClient} is responsible for endpoint of Consul API:
+ * {@code `/health`}(https://www.consul.io/api/health.html)
+ *
+ * <p>GET /health/node/:node
+ * GET /health/checks/:service
+ * GET /health/service/:service?passing
+ * GET /health/connect/:service
+ * GET /health/state/:state
+ */
+public class HealthClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HealthClient.class);
+
+    private final ConsulClient client;
+
+    public static HealthClient of(ConsulClient consulClient) {
+        return new HealthClient(consulClient);
+    }
+
+    protected HealthClient(ConsulClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Gets a list of nodes for service.
+     */
+    private CompletableFuture<List<HealthService>> service(String serviceName, ConsulParameters parameters) {
+        logger.trace("service() {}, {}", serviceName, parameters.toParameterString());
+        return client.get("/health/service/" + serviceName + '?' + parameters.toParameterString())
+                     .aggregate()
+                     .thenApply(response -> {
+                         final List<HealthService> healths = new ArrayList<>();
+                         logger.trace("response: {}", response);
+                         if (!Objects.equals(response.status(), HttpStatus.OK)) {
+                             return healths;
+                         }
+                         final ObjectMapper mapper = new ObjectMapper();
+                         final HealthService[] resultHealthServices;
+                         try {
+                             resultHealthServices =
+                                     mapper.readValue(response.content().toStringUtf8(), HealthService[].class);
+                         } catch (IOException e) {
+                             return healths;
+                         }
+                         if (resultHealthServices == null) {
+                             return healths;
+                         }
+                         healths.addAll(Arrays.asList(resultHealthServices));
+                         return healths;
+                     });
+    }
+
+    /**
+     * Gets a endpoint list with service name and parameter.
+     */
+    public CompletableFuture<List<Endpoint>> endpoints(String serviceName, ConsulParameters parameters) {
+        return service(serviceName, parameters).thenApply(nodes -> {
+            final List<Endpoint> endpoints = new ArrayList<>(nodes.size());
+            nodes.forEach(node -> {
+                final String host = node.service.address.isEmpty() ?
+                                    node.node.address.isEmpty() ? "127.0.0.1"
+                                                                : node.node.address
+                                                                   : node.service.address;
+                endpoints.add(Endpoint.of(host, node.service.port));
+            });
+            return endpoints;
```

Comment:
nit: Could use stream and how about using `if ...else` than nested ternary operator?
```java
return nodes.stream()
            .map(node -> {
                final String host;
                if (!node.service.address.isEmpty()) {
                    host = node.service.address;
                } else if (!node.node.address.isEmpty()) {
                    host = node.node.address;
                } else {
                    host = "127.0.0.1";
                }
                return Endpoint.of(host, node.service.port);
            }).collect(toImmutableList());
```